refactor: Group and re-order CAddrMan members by access type

hebasto opened the pull request:
This PR is split from #19238 as all of its commits are trivial to review.
The last commit is easy to review with `git diff --color-moved=dimmed-zebra`.

Addressed the following comments from #19238:
- https://github.com/bitcoin/bitcoin/pull/19238/commits/130b82ff3588d7207c928bda16b96e720e0b24cd#r550865131
> Can you consolidate all the private members and protected members to be next to each other? Multiple private and protected access specifiers make this harder to read than is necessary.

- https://github.com/bitcoin/bitcoin/pull/19238/commits/130b82ff3588d7207c928bda16b96e720e0b24cd#r557271783
> Yeah, class declaration is easier to read if there is just one instance of `public:`, `protected:` and `private:` (in that order).

hebasto committed:
refactor: Do not expose CAddrMan members as protected without need

vasild commented on src/addrman.h:
```diff
@@ -759,6 +626,141 @@ friend class CAddrManTest;
+    friend class CAddrManCorrupted;
+    friend class CAddrManDeterministic;
+    friend class CAddrManSerializationMock;
+    friend class CAddrManTest;
+
 };
```
nit: this empty line can be removed

hebasto replied:
Thanks! Done.

vasild reviewed (approved):
ACK 6627b0bab44a6e8392611182fdec4c4a85ba10a0

vasild reviewed (approved):
ACK 1b177a442a2e3749073cc189ddd404e74a902923

jnewbery commented:
ACK 1b177a442a2e3749073cc189ddd404e74a902923

laanwj commented on src/addrman.h:
```diff
@@ -759,6 +626,140 @@ friend class CAddrManTest;
+    //! Update an entry's service bits.
+    void SetServices_(const CService &addr, ServiceFlags nServices) EXCLUSIVE_LOCKS_REQUIRED(cs);
+
+    friend class CAddrManCorrupted;
```
Sorry but I'm not 100% sure that having this class know about all descendents in the tests is better than using `protected`.

hebasto replied:
Currently, one derived testing class has the `friend` specifier, others have access to `protected` members.
Is combining both approaches is better?

laanwj replied:
I'd say that if there is a choice, using `protected` is better than explicitly naming every descendant. Ideally you'd be able to add tests without having to mention them in the code to be tested. `friend` is like a last resort option for otherwise unrelated classes.

hebasto replied:
I see.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/22025#issuecomment-847180767).

hebasto committed:
move-only: Group and re-order CAddrMan members by access type

Easy to verify with `git diff --color-moved=dimmed-zebra`.

hebasto commented:
Updated 1b177a442a2e3749073cc189ddd404e74a902923 -> 8caf60dbbe16afa3c52574a7f6710d74c0bfd4ab ([pr22025.02](https://github.com/hebasto/bitcoin/commits/pr22025.02) -> [pr22025.03](https://github.com/hebasto/bitcoin/commits/pr22025.03), [diff](https://github.com/hebasto/bitcoin/compare/pr22025.02..pr22025.03)):

- addressed @laanwj's [comment](https://github.com/bitcoin/bitcoin/pull/22025#discussion_r637977808)

jnewbery commented:
ACK 8caf60dbbe16afa3c52574a7f6710d74c0bfd4ab

practicalswift commented:
Concept ACK

jarolrod reviewed (commented):
ACK 8caf60dbbe16afa3c52574a7f6710d74c0bfd4ab

Patch looks correct

vasild reviewed (approved):
ACK 8caf60dbbe16afa3c52574a7f6710d74c0bfd4ab

laanwj commented:
Code review ACK 8caf60dbbe16afa3c52574a7f6710d74c0bfd4ab
